redis/tests/test_cluster_async.rs

```diff
@@ -380,52 +380,60 @@ fn test_async_cluster_tryagain_exhaust_retries() {
 fn test_async_cluster_rebuild_with_extra_nodes() {
     let name = "rebuild_with_extra_nodes";
 
-    let requests = atomic::AtomicUsize::new(0);
+    let requests = Arc::new(AtomicI32::new(0));
     let started = atomic::AtomicBool::new(false);
     let MockEnv {
         runtime,
         async_connection: mut connection,
         handler: _handler,
         ..
-    } = MockEnv::new(name, move |cmd: &[u8], port| {
-        if !started.load(atomic::Ordering::SeqCst) {
-            respond_startup(name, cmd)?;
-        }
-        started.store(true, atomic::Ordering::SeqCst);
+    } = MockEnv::new(name, {
+        let i = requests.clone();
+        move |cmd: &[u8], port| {
+            if !started.load(atomic::Ordering::SeqCst) {
+                respond_startup_two_nodes(name, cmd)?;
+            }
+            started.store(true, atomic::Ordering::SeqCst);
 
-        if contains_slice(cmd, b"PING") {
-            return Err(Ok(Value::Status("OK".into())));
-        }
+            if contains_slice(cmd, b"PING") {
+                return Err(Ok(Value::Status("OK".into())));
+            }
 
-        let i = requests.fetch_add(1, atomic::Ordering::SeqCst);
-
-        match i {
-            // Respond that the key exists elswehere (the slot, 123, is unused in the
-            // implementation)
-            0 => Err(parse_redis_value(b"-MOVED 123\r\n")),
-            // Respond with the new masters
-            1 => Err(Ok(Value::Bulk(vec![
-                Value::Bulk(vec![
-                    Value::Int(0),
-                    Value::Int(1),
+            let i = i.fetch_add(1, atomic::Ordering::SeqCst);
+
+            match i {
+                // Respond that the key exists elswehere (the slot, 123, is unused in the
+                // implementation)
+                0 => Err(parse_redis_value(
+                    format!("-MOVED 14000 {name}:6380\r\n").as_bytes(),
+                )),
+                1 => {
+                    // Check that the correct node receives the request after rebuilding
```

Comment:
this happens before rebuilding, according to the atomic, no?

Reply:
I believe this comment is no longer relevant, but please confirm.

Reply:
you're right.